src/porepy/models/constitutive_laws.py

```diff
@@ -2870,20 +2870,10 @@ def elastic_normal_fracture_deformation(
         # If the maximum closure is zero, the Barton-Bandis model is not valid in the
         # case of zero normal traction. In this case, we return an empty operator.
         #  If the maximum closure is negative, an error is raised.
-        val = maximum_closure.evaluate(self.equation_system)
-        if (
-            (isinstance(val, (float, int)) and val == 0)
-            or (isinstance(val, np.ndarray) and np.any(val == 0))
-            or isinstance(val, pp.ad.AdArray)
-            and np.any(val.val == 0)
-        ):
+        val = maximum_closure.value(self.equation_system)
+        if np.any(val == 0):
```

Comment:
Should we assert that val is numpy array or a numbers.real here?

Reply:
Decision: ignore it.